src/Bicep.LangServer.IntegrationTests/TelemetryTests.cs

```diff
@@ -0,0 +1,142 @@
+            IDictionary<string, string> properties = new Dictionary<string, string>
+            {
+                { "name", "{}" }
+            };
+
+            TaskCompletionSource<TelemetryEventParams> telemetryReceived = await ResolveCompletionAsync(text, "{}", TelemetryConstants.EventNames.ModuleBodySnippetInsertion, properties);
+            TelemetryEventParams telemetryEventParams = await IntegrationTestHelper.WithTimeoutAsync(telemetryReceived.Task);
+            IDictionary<string, object> extensionData = telemetryEventParams.ExtensionData;
+            ICollection<string> keys = extensionData.Keys;
+
+            keys.Count.Should().Be(2);
+            keys.Should().Contain("eventName");
+            keys.Should().Contain("properties");
+            extensionData["eventName"].ToString().Should().Be(TelemetryConstants.EventNames.ModuleBodySnippetInsertion);
+            extensionData["properties"].ToString().Should().BeEquivalentToIgnoringNewlines(@"{
+  ""name"": ""{}""
+}");
+        }
+
+        private async Task<TaskCompletionSource<TelemetryEventParams>> ResolveCompletionAsync(string text, string prefix, string eventName, IDictionary<string, string> properties)
+        {
+            var fileSystemDict = new Dictionary<Uri, string>();
+            var telemetryReceived = new TaskCompletionSource<TelemetryEventParams>();
+
+            var client = await IntegrationTestHelper.StartServerWithClientConnectionAsync(
+                TestContext,
+                options =>
+                {
+                    options.OnTelemetryEvent(telemetry => telemetryReceived.SetResult(telemetry));
+                },
+                fileResolver: new InMemoryFileResolver(fileSystemDict));
+
+            var mainUri = DocumentUri.FromFileSystemPath("/main.bicep");
+            fileSystemDict[mainUri.ToUri()] = text;
+
+            client.TextDocument.DidOpenTextDocument(TextDocumentParamHelper.CreateDidOpenDocumentParams(mainUri, fileSystemDict[mainUri.ToUri()], 1));
+
+            var completions = await client.RequestCompletion(new CompletionParams
+            {
+                TextDocument = new TextDocumentIdentifier(mainUri),
+                Position = new Position(0, text.Length),
+            });
+
+            CompletionItem completionItem = completions.Where(x => x.Kind == CompletionItemKind.Snippet && x.Label == prefix).First();
+            Command? command = completionItem.Command;
+
+            command!.Name.Should().Be(TelemetryConstants.CommandName);
+
+            JArray? arguments = command!.Arguments;
+            BicepTelemetryEvent? telemetryEvent = arguments!.First().ToObject<BicepTelemetryEvent>();
+
+            telemetryEvent!.EventName.Should().Be(eventName);
+            telemetryEvent!.Properties!.Should().Equal(properties);
+
+            await client.ResolveCompletion(completionItem);
+            await client.Workspace.ExecuteCommand(command);
+
+            return telemetryReceived;
```

Comment:
Why not just return the data directly to avoid repetition:
```csharp
await IntegrationTestHelper.WithTimeoutAsync(telemetryReceived.Task);
```

Reply:
Updated here:
https://github.com/Azure/bicep/commit/62d6763f06b3d53bb8189b517dab0567a37fd0bd8